node/building_blocks.re

```diff
@@ -96,18 +119,62 @@ let is_signable = (state, block) => {
   && is_current_producer(state, ~key_hash=block.author)
   && !has_next_block_to_apply(state, ~hash=block.hash)
   && all_main_ops_are_known
-  && contains_only_trusted_add_validator_op(block.side_chain_ops);
+  && contains_only_trusted_add_validator_op(block.side_chain_ops)
+  && block_has_signable_state_root_hash(~current_time, state, block);
 };
 
 let sign = (~key, block) => Block.sign(~key, block);
 
-let produce_block = state =>
+/** The state_root_min_timeout determines how often the block producer
+    starts hashing a new state root. It is closely related to the
+    state root epoch (see notes in [should_start_new_epoch]]).
+
+    The faster hashing is, the shorter this time should be. Faster updates
+    to the state root hash means shorter times to join the network, since a
+    node only needs to download and apply the blocks from the previous state root
+    update to the current block.
+*/
+let state_root_min_timeout = 60.0;
+
+/** Calculates whether to start sending a new state root hash.
+
+    The state root epoch is the interval (in blocks) between state root
+    hash updates. Thus, a new epoch is triggered by applying a block with
+    a new state root hash. The block producer decides when to send
+    blocks with new state root hashes. To enforce that he does so on time,
+    validators reject blocks with updates that occur to soon or too late
+    (see [Protocol.apply]).
```

Comment:
The validator still applies the block if needed, he just doesn't sign it.

Probably not this function, I believe it will be `Flows.is_signable`.